OpenRA.Mods.Common/Traits/UpgradesIcons.cs

```diff
@@ -0,0 +1,137 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Attach this to the player actor. When attached, enables all actors possessing the UpgradesWhenCreated ",
+		"trait to have their production queue icons render with an overlay defined in this trait. ",
+		"The icon change occurs when UpgradesWhenCreated.Prerequisites are met.")]
+	public class UpgradesIconsInfo : ITraitInfo, Requires<TechTreeInfo>
+	{
+		[Desc("Image used for this decoration. Defaults to the actor's type.")]
```

Comment:
I've done a copy-paste from `WithDecoration` and forgot to adjust the descriptions. I'll fix it in a minute.